src/NewNode.java

```diff
@@ -0,0 +1,17 @@
+public class NewNode<T> {
```

Comment:
better make it private static nested classes

Reply:
changed everything according to the comments